addon-test-support/utils.js

```diff
@@ -1,5 +1,54 @@
+import Ember from 'ember';
+
+/**
+ * Utility to check performance API.
+ * 
+ * @return {Boolean}
+ * @private
+ */
+function hasPerformanceApi() {
+  return window &&
+    typeof window.performance !== 'undefined' &&
+    typeof window.performance.mark === 'function' &&
+    typeof window.performance.measure === 'function';
+}
+
 export default {
+  /**
+   * Exposes performance API if it exists.
+   *
+   * @constant
+   * @public
+   * @type {Object}
+   */
+  performance: hasPerformanceApi() ? window.performance : undefined,
+
+  /**
+   * Utility to get window location object.
+   *
+   * @return {Object}
+   * @public
+   */
   getLocation() {
     return window && window.location;
+  },
+
+  /**
+   * Utility to measure performance between start and end markers.
+   *
+   * @param {String} comment
+   * @param {String} startMark
+   * @param {String} endMark
+   * @return {Void}
+   * @public
+   */
+  measure(comment, startMark, endMark) {
+    // `performance.measure` may fail if the mark could not be found.
+    // reasons a specific mark could not be found include outside code invoking `performance.clearMarks()`
+    try {
+      performance.measure(comment, startMark, endMark);
+    } catch (e) {
+      Ember.Logger.warn('performance.measure could not be executed because of ', e.message);
```

Comment:
Ember.Logger is deprecated (I think the deprecation was added in 3.1 or 3.2), we should swap to `console.warn`.

See [the deprecation guide](https://emberjs.com/deprecations/v3.x#toc_ember-console-deprecate-logger) for more info.

Reply:
Cool, will do!

Reply:
Fixed.